creator-node/src/ipfsClient.js

```diff
@@ -15,11 +19,100 @@ const ipfsLatest = ipfsClientLatest({ host: ipfsAddr, port: config.get('ipfsPort
+  genericLogger.info(`Current IPFS Peer ID (using latest version of ipfs client): ${JSON.stringify(identityLatest, null, 2)}`)
+}
+
+/**
+ * Wrapper to ipfs.addToFs() -- This wrapper allows for ipfs calls to be made async or synchronous.
+ * @param {Object} ipfs ipfs library -- can be two different versions (see ipfsClient.js)
+ * @param {Buffer|Buffer[]} buffers
+ * @param {Object?} ipfsConfig ipfs add config options
+ * @param {string?} logContext the name of the caller fn
+ * @param {boolean?} addToIPFSDaemon flag to override adding to ipfs daemon
+ * @returns {string|string[]} hashes
+ */
+async function ipfsAddWrapper (ipfs, buffers, ipfsConfig = {}, logContext = {}, addToIPFSDaemon = false) {
+  const logger = genericLogger.child(logContext)
+
+  // If async ipfs add is enabled, or if `addToIPFSDaemon` flag passed in is false, asynchronously
+  // add to ipfs. Else, synchronously add to ipfs and wait for the response.
+  let ipfsDaemonHashes
+  if (ENABLE_ASYNC_IPFS_ADD && !addToIPFSDaemon) {
+    ipfs.add(buffers, ipfsConfig) // Do not await it
+  } else {
+    ipfsDaemonHashes = await ipfs.add(buffers, ipfsConfig)
+  }
+
+  // If `buffers` is an array, get the content hash of each element. Else, it will be a single element.
+  // Also, if `buffers` is a single element, get the single element hash from `ipfsDaemonHashes`
+  let onlyHashes = []
+  if (Array.isArray(buffers)) {
+    // Note: See ipfs add code in `resizeImage.js` for `buffers` structure
+    for (const { content } of buffers) {
+      const hash = await ipfsHashOf(content)
+      onlyHashes.push(hash)
+    }
+  } else {
+    onlyHashes = await ipfsHashOf(buffers)
+    ipfsDaemonHashes = ipfsDaemonHashes[0].hash
+  }
+
+  const ipfsDaemonHashLogStr = `ipfsDaemonHash=${ipfsDaemonHashes} isSameHash=${onlyHashes === ipfsDaemonHashes}`
+  logger.info(`[ipfsClient - ipfsAddWrapper()] onlyHash=${onlyHashes} ${ipfsDaemonHashes ? ipfsDaemonHashLogStr : ''}`)
+
+  // If content was added to ipfs daemon, prioritize using that hash response
+  return ipfsDaemonHashes || onlyHashes
+}
+
+/**
+ * Wrapper to ipfs.add() -- This wrapper allows for ipfs calls to be made async or synchronous.
+ * @param {Object} ipfs ipfs library -- can be two different versions (see ipfsClient.js)
+ * @param {string|string[]} srcPath
+ * @param {Object?} ipfsConfig ipfs add config options
+ * @param {string?} logContext request context; used to log more info on caller
+ * @returns {string|string[]} hashes
+ */
+async function ipfsAddToFsWrapper (ipfs, srcPath, ipfsConfig = {}, logContext = {}) {
+  const logger = genericLogger.child(logContext)
+
+  const stream = fs.createReadStream(srcPath)
+  const onlyHash = await ipfsHashOf(stream)
+  if (ENABLE_ASYNC_IPFS_ADD) {
+    logger.info(`[ipfsClient - ipfsAddToFsWrapper()] onlyHash=${onlyHash}`)
+    ipfs.addFromFs(srcPath, ipfsConfig) // Do not await it
+    return onlyHash
+  }
```

Comment:
in this case we don't get the isSameHash log like we do in ipfsAddWrapper above - def want to keep that consistent in both and make sure we always have that log for verification. also in the above function we always return the hash from ipfs daemon, why not do that here?

Reply:
we actually don't return the ipfs daemon hash all the time. if the async ipfs add is enabled, we only do content hashing. the ipfs daemon hash may be null at times, and that's why I decided to not include that in the log